api/src/paths/project/{projectId}/survey/{surveyId}/devices/delete.ts

```diff
@@ -0,0 +1,136 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { PROJECT_PERMISSION, SYSTEM_ROLE } from '../../../../../../constants/roles';
+import { getDBConnection } from '../../../../../../database/db';
+import { authorizeRequestHandler } from '../../../../../../request-handlers/security/authorization';
+import { TelemetryDeviceService } from '../../../../../../services/telemetry-services/telemetry-device-service';
+import { getLogger } from '../../../../../../utils/logger';
+
+const defaultLog = getLogger('paths/project/{projectId}/survey/{surveyId}/devices2/delete');
+
+export const POST: Operation = [
+  authorizeRequestHandler((req) => {
+    return {
+      or: [
+        {
+          validProjectPermissions: [PROJECT_PERMISSION.COORDINATOR, PROJECT_PERMISSION.COLLABORATOR],
+          surveyId: Number(req.params.surveyId),
+          discriminator: 'ProjectPermission'
+        },
+        {
+          validSystemRoles: [SYSTEM_ROLE.DATA_ADMINISTRATOR],
+          discriminator: 'SystemRole'
+        }
+      ]
+    };
+  }),
+  deleteDevices()
+];
+
+POST.apiDoc = {
+  description: 'Delete devices.',
+  tags: ['device'],
+  security: [
+    {
+      Bearer: []
+    }
+  ],
+  parameters: [
+    {
+      in: 'path',
+      name: 'projectId',
+      schema: {
+        type: 'integer',
+        minimum: 1
+      },
+      required: true
+    },
+    {
+      in: 'path',
+      name: 'surveyId',
+      schema: {
+        type: 'integer',
+        minimum: 1
+      },
+      required: true
+    }
+  ],
+  requestBody: {
```

Comment:
suggestion: We can include `required: true` to prevent empty request bodies.

Reply:
We definitely should, good catch. I hadn't even considered that the whole body could be optional/marked required. Its probably missing in a ton of endpoints.

Reply:
I might open a separate PR for that then

Reply:
Added a JIRA ticket: https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-630

Reply:
Yea, I just caught it yesterday when I was testing my endpoint. Without it, it allows you to get pretty far into the endpoint before it throws.